testcases/MachineConfig.py

```diff
@@ -0,0 +1,259 @@
+                curr_num_volumes = self.cv_HMC.vpmem_count()
+                if curr_num_volumes[0] >= "1":
+                    log.info("Configured vpmem %s of %sMB" % (self.pmem_name, self.pmem_size))
+                else:
+                    self.fail("Failed to configure pmem")
+
+
+        self.cv_HMC.run_command("chsysstate -r lpar -m %s -o on -n %s -f %s" %
+                               (self.system_name, self.lpar_name, self.lpar_prof))
+        time.sleep(10)
+        curr_proc_mode = self.cv_HMC.get_proc_mode()
+        if proc_mode in curr_proc_mode:
+            log.info("System booted with %s mode" % proc_mode)
+        else:
+            self.fail("Failed to boot in %s mode" % proc_mode)
+
+
+class RestoreLAPRConfig(MachineConfig):
+
+    '''
+    This class boots the lpar with previous profile, this can be called after ChangeToSharedMode or ChangeToDedicatedMode.
+    '''
+
+    def setUp(self):
+        super(RestoreLAPRConfig, self).setUp()
+        conf = OpTestConfiguration.conf
+
+    def runTest(self):
+        self.cv_HMC.poweroff_lpar()
+        self.cv_HMC.lpar_prof = self.cv_HMC.lpar_prof + "_bck"
+        self.cv_HMC.poweron_lpar()
+
+
+class CecConfig(MachineConfig):
+
+    '''
+    This class configures LMB. Pass lmb_size to cec_config in config file.
+    Ex: cec_config="lmb=4GB, hugepage=16GB"
+    Valid LMB values are "128,256,1024,2048,4096"
+    '''
+
+    def setUp(self):
+        super(CecConfig, self).setUp()
+        conf = OpTestConfiguration.conf
+        try: self.cec_config = conf.args.cec_config
+        except AttributeError:
+            self.cec_config="lmb=256"
+
+    def runTest(self):
+        if "lmb=4GB" in self.cec_config or "lmb=4096" in self.cec_config:
+            self.lmb_size=4096
+        elif "lmb=2GB" in self.cec_config or "lmb=2048" in self.cec_config:
+            self.lmb_size=2048
+        elif "lmb=1GB" in self.cec_config or "lmb=1024" in self.cec_config:
+            self.lmb_size=1024
+        elif "lmb=256MB" in self.cec_config or "lmb=256" in self.cec_config:
+            self.lmb_size=256
+        elif "lmb=128MB" in self.cec_config or "lmb=128" in self.cec_config:
+            self.lmb_size=128
+        else:
+            self.skipTest("Please pass valid LMB size in config file ex: 128,256,1024,2048,4096")
+        current_lmb = self.cv_HMC.get_lmb_size()
+        if int(current_lmb[0]) == int(self.lmb_size):
+            log.info("System is already booted with LMB %s" % self.lmb_size)
+        else:
+            self.cv_HMC.configure_lmb(self.lmb_size)
+            self.cv_HMC.poweroff_system()
+            self.cv_HMC.poweron_system()
+            current_lmb = self.cv_HMC.get_lmb_size()
+            if int(current_lmb[0]) == int(self.lmb_size):
+                log.info("System booted with LMB %s" % self.lmb_size)
+            else:
+                self.fail("Failed to boot with LMB %s" % self.lmb_size)
+            if self.cv_HMC.lpar_vios:
```

Comment:
this should be first step as if set it LMB size and vios is not given operation is make race condition